api_app/playbooks_manager/controller.py

```diff
@@ -0,0 +1,283 @@
+# This file is a part of IntelOwl https://github.com/intelowlproject/IntelOwl
+# See the file 'LICENSE' for copying permission.
+
+import logging
+from typing import Dict, List
+
+from celery import chord, uuid
+from django.conf import settings
+from api_app.analyzers_manager.dataclasses import AnalyzerConfig
+from api_app.connectors_manager.dataclasses import ConnectorConfig
+
+from api_app.exceptions import AlreadyFailedJobException, NotRunnableAnalyzer,  \
+    NotRunnableConnector, NotRunnablePlaybook
+
+from api_app.helpers import get_now
+from ..models import Job
+from intel_owl.consts import DEFAULT_QUEUE
+
+from .dataclasses import PlaybookConfig
+
+logger = logging.getLogger(__name__)
+
+
+def filter_playbooks(serialized_data: Dict, warnings: List[str]) -> List[str]:
+    # init empty list
+    cleaned_playbook_list = []
+    selected_playbooks = []
+    analyzers_to_be_run = []
+    connectors_to_be_run = []
+
+    # get values from serializer
+    playbooks_requested = serialized_data[0].get("playbooks_requested", [])
+
+    # read config
+    playbook_dataclasses = PlaybookConfig.all()
+    all_playbook_names = list(playbook_dataclasses.keys())
+
+    # run all playbooks; Might just remove this for now.
+    run_all = len(playbooks_requested) == 0
+    if run_all:
+        # select all
+        selected_playbooks.extend(all_playbook_names)
+    else:
+        # select the ones requested
+        selected_playbooks.extend(playbooks_requested)
+        
+    for p_name in selected_playbooks:
+        try:
+            pp = playbook_dataclasses.get(p_name, None)
+            analyzers_to_be_run.extend(pp.analyzers)
+            connectors_to_be_run.extend(pp.connectors)
+            if not pp:
+                if not run_all:
+                    raise NotRunnablePlaybook(
+                        f"{p_name} won't run: not available for configuration"
```

Comment:
`not present in configuration` maybe?

Reply:
My bad. That would be a better log to store.